File: ZezeJava/ZezeJava/src/main/java/Zeze/Raft/RocksRaft/LogList.java
package Zeze.Raft.RocksRaft;

import org.pcollections.PVector;

public abstract class LogList<V> extends LogBean {
	private PVector<V> value;

	public LogList(int typeId) {
		super(typeId);
	}

	public LogList(String typeName) {
		super(typeName);
	}

	final PVector<V> getValue() {
		return value;
	}

	final void setValue(PVector<V> value) {
		this.value = value;
	}

	@Override
	public void collect(Changes changes, Bean recent, Log vlog) {
		throw new UnsupportedOperationException("Collect Not Implement.");
	}
}

File: Zeze/Raft/RocksRaft/LogList.cs

using System;
using System.Collections.Immutable;

namespace Zeze.Raft.RocksRaft
{
	public abstract class LogList<E> : LogBean
	{
		internal ImmutableList<E> Value { get; set; }

		public override void Collect(Changes changes, Bean recent, Log vlog)
		{
			throw new NotImplementedException("Collect Not Implement.");
		}
	}
}
